File: java-home-inventory-core/src/br/com/sailboat/homeinventory/core/interactor/validator/ProductValidator.java
package br.com.sailboat.homeinventory.core.interactor.validator;

import br.com.sailboat.homeinventory.core.entity.Product;
import br.com.sailboat.homeinventory.core.exception.FieldNotFilledException;

public class ProductValidator {

    private Product product;

    public ProductValidator(Product product) {
        this.product = product;
    }

    public static void validate(Product product) throws FieldNotFilledException {
        ProductValidator validator = new ProductValidator(product);
        validator.validateName();
        validator.validateQuantity();
    }

    public void validateName() throws FieldNotFilledException {
        if (product.getName() == null || product.getName().trim().isEmpty()) {
            throw new FieldNotFilledException();
        }
    }

    public void validateQuantity() throws FieldNotFilledException {
        if (product.getQuantity() != null && product.getQuantity() < 0) {
            throw new FieldNotFilledException();
        }
    }
}

File: android-home-inventory/app/src/main/java/br/com/sailboat/homeinventory/interactor/validator/ProductValidator.kt
package br.com.sailboat.homeinventory.interactor.validator

import android.content.Context
import br.com.sailboat.canoe.exception.RequiredFieldNotFilledException
import br.com.sailboat.canoe.helper.StringHelper
import br.com.sailboat.homeinventory.R
import br.com.sailboat.homeinventory.model.Product

class ProductValidator {

    companion object {

        fun validate(context: Context, product: Product) {
            if (StringHelper.isNullOrEmpty(product.name)) {
                throw RequiredFieldNotFilledException(context.getString(R.string.exception_product_name))
            }

        }

    }
}